embassy-stm32/src/usart/mod.rs

```diff
@@ -518,19 +518,6 @@ impl<'d, T: BasicInstance, RxDma> UartRx<'d, T, RxDma> {
             unreachable!();
         }
 
-        if enable_idle_line_detection {
-            // clear idle flag
-            let sr = sr(r).read();
-            // This read also clears the error and idle interrupt flags on v1.
-            unsafe { rdr(r).read_volatile() };
-            clear_interrupt_flags(r, sr);
-
-            // enable idle interrupt
-            r.cr1().modify(|w| {
-                w.set_idleie(true);
-            });
-        }
-
```

Comment:
I'm not sure this code needs to be removed. If errors exist before the method is called, then they need to be cleared.

Reply:
1. If errors exist, it will be cleared correctly, the code is located in line 500~503
2. There is no difference when `enable_idle_line_detection` is false
3. When `enable_idle_line_detection` is true, line 529~523 will do the same thing such as  the code I removed.